custom-routes-vis/src/main/webapp/script.js

```diff
@@ -137,13 +138,98 @@ function showMarker(markerName) {
           document.getElementById('generate-routes').style.display = 'block';
         }
       });
-
+  document.getElementById(markerName + '-lat').style.display =
+      'none';
+  document.getElementById(markerName + '-lng').style.display =
+      'none';
+  document.getElementById('show-' + markerName + '-custom-marker').style.display =
+      'none';
   document.getElementById('show-' + markerName + '-marker').style.display =
       'none';
   document.getElementById(markerName + '-coordinates').innerHTML =
       'Click the map to select location!';
 }
 
+/**
+ * Hides 'Submit Custom Coordinates' and 'Place Marker' button and waits for user
+ * to input coordinates for the marker. Updates marker to be in correct location,
+ * and show 'Delete Marker' button.
+ * @param {string} markerName Name of corresponding marker in
+ *     originDestinationMarkers array.
+ */
+function showCustomCoordinatesMarker(markerName) {
+  let containerOfOtherMarkerName;
+  let markerIndex;
+  switch (markerName) {
+    case MarkerNames.ORIGIN:
+      containerOfOtherMarkerName =
+          'hide-' + MarkerNames.DESTINATION + '-marker';
+      markerIndex = 0;
+      break;
+    case MarkerNames.DESTINATION:
+      containerOfOtherMarkerName = 
+          'hide-' + MarkerNames.ORIGIN + '-marker';
+      markerIndex = 1;
+      break;
+  }
+  let customLat = parseFloat(document.getElementById(markerName + '-lat').value);
+  let customLng = parseFloat(document.getElementById(markerName + '-lng').value);
+  if(!checkCustomCoordinatesInputIsValid(customLat, customLng)){
+    return;
+  }
+  let customLatLng = new google.maps.LatLng({lat: customLat, lng: customLng}); 
+  originDestinationMarkers[markerIndex].setPosition(customLatLng);
```

Comment:
Can you also set the center of the map to be this position? This ensures the user can see the marker.

Reply:
+1. If we center the map to this position, there is no need to change the zoom level to 3. And it's ok to center the map twice when you input the origin and destination.

Reply:
+1